### PR TITLE
feat: reintroduce usernames

### LINKED ISSUE #422: Reintroduce usernames
Needed for https://github.com/Sirimangalo/meditation-plus-angular/issues/58 and it's useful in general. 
Ideas:
- We should ask for the username on the first login and invalidate every session without a username.
- Usernames shouldn't be allowed to change (for mentions).

### PR DESCRIPTION
**Pull request by binarious**
Depends on https://github.com/Sirimangalo/meditation-plus-server/pull/136

Closes #422

### CODE REVIEW

File: src/app/login/login.component.html [hunk 1]
```diff
@@ -12,10 +12,19 @@
...
               #mail="ngModel"
               [(ngModel)]="email"
-              type="email"
-              placeholder="Email"
+              placeholder="Email or Username"
               required
-              dividerColor="(mail.valid || mail.pristine) ? 'primary' : 'warn'">
+              dividerColor="(mail.valid || mail.pristine) ? 'primary' : 'warn'"
+              *ngIf="!doSignup">
+              <input
```
Comment (olastor) [inline src/app/login/login.component.html]: Is there a reason for this second input?

File: src/app/admin/analytics/analytics.component.html [hunk 2]
```diff
@@ -36,7 +36,7 @@
           <span>Nothing found</span>
         </div>
         <md-list *ngIf="users && users.admins">
-          <a *ngFor="let user of users.admins" class="online-row" [routerLink]="['/profile', user._id]">
+          <a *ngFor="let user of users.admins" class="online-row" [routerLink]="['/profile', user.username]">
```
Comment (olastor) [inline src/app/admin/analytics/analytics.component.html]: Any profile link of an user that has not set his username yet, will route to `/profile/undefined`.  Is that acceptable?

File: src/app/user/user.service.ts [hunk 3]
```diff
@@ -26,11 +26,13 @@ export class UserService {
    * Logging in a User via email and password.
    * @param {String} email
    * @param {String} password
+   * @param {String} username
    */
-  public login(email: string, password: string) {
+  public login(email: string, password: string, username: string = undefined) {
+    console.log('username', username);
```
Comment (olastor) [inline src/app/user/user.service.ts]: is this log intentional?

File: src/app/login/login.component.html (hunk 1 above)
Comment (binarious) [inline src/app/login/login.component.html]: Yes, the positioning in the form and the placeholder.

File: src/app/user/user.service.ts (hunk 3 above)
Comment (binarious) [inline src/app/user/user.service.ts]: Nope, needs to be removed. Thank you.

File: src/app/admin/analytics/analytics.component.html (hunk 2 above)
Comment (binarious) [inline src/app/admin/analytics/analytics.component.html]: Was thinking about this, too. The only acceptable solution would be:
```html
[routerLink]="user.username ? ['/profile', user.username] : ['/profile/id', user._id]"
```
I think, I'll fix this.

**Comment by binarious** [comment]
@olastor Things are fixed.

**Pull request merged**